src/app/shared/guards/pending-changes-guard.ts

```diff
@@ -44,16 +46,26 @@ export class PendingChangesGuard implements CanDeactivate<PendingChangesComponen
       icon: 'pi pi-exclamation-triangle',
       acceptLabel: $localize`Yes, leave page`,
       accept: () => {
-        dialogResponse.next(true);
-        dialogResponse.complete();
+        dialogResponse.next('accepted');
       },
       rejectLabel: $localize`No`,
       reject: () => {
-        dialogResponse.next(false);
-        dialogResponse.complete();
+        dialogResponse.next('declined');
       },
     });
-    return dialogResponse;
+
+    dialogResponse$.pipe(
```

Comment:
if the goal of the `dialogResponse$` variable is only to be used here, just use `dialogResponse` here, that's the same.